app/models/nic/managed.rb

```diff
@@ -78,6 +95,46 @@ def dhcp_attrs
       }
     end
 
+    def copy_hostname_from_host
+      self.name = self.host.read_attribute :name
+    end
+
+    def set_provisioning_flag
+      if self.primary? && self.host && self.host.interfaces.detect(&:provision).nil?
+        self.provision = true
+      end
+    end
+
+    def update_lookup_value_fqdn_matchers
+      if self.primary? && self.fqdn_changed?
+        LookupValue.where(:match => "fqdn=#{fqdn_was}").update_all(:match => host.send(:lookup_value_match))
+      end
+    end
+
+    def drop_host_cache
+      if self.host
+        self.host.drop_primary_interface_cache if self.primary
+        self.host.drop_provision_interface_cache if self.provision
+      end
+      true
+    end
+
+    # log errors to host object since we can't read it later (even if host.destroy fails host.interfaces is
+    # always set to [] so we lose interfaces errors)
+    def failure(msg, backtrace = nil, dest = :base)
+      result = super
+      host.errors.add(dest, msg)
```

Comment:
Be more consistent when using ```self.```, other additions in the same class use it but not this method. I'm all for not using ```self``` unless required but let's try to be consistent. ```ip_available``` and ```mac_available``` also don't use it so either remove it everywhere else or add it here, preferably the former?